Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2300,10 +2301,22 @@ - (void)showAttachmentDocumentPickerMenu
     // UIDocumentPickerModeImport copies to a temp file within our container.
     // It uses more memory than "open" but lets us avoid working with security scoped URLs.
     UIDocumentPickerMode pickerMode = UIDocumentPickerModeImport;
+    // TODO: UIDocumentMenuViewController is deprecated; we should use UIDocumentPickerViewController
+    //       instead.
     UIDocumentMenuViewController *menuController =
         [[UIDocumentMenuViewController alloc] initWithDocumentTypes:documentTypes inMode:pickerMode];
     menuController.delegate = self;
 
+    UIImage *takeMediaImage = [UIImage imageNamed:@"actionsheet_camera_black"];
+    OWSAssert(takeMediaImage);
+    [menuController addOptionWithTitle:NSLocalizedString(
+                                           @"MEDIA_FROM_LIBRARY_BUTTON", @"media picker option to choose from library")
+                                 image:takeMediaImage
+                                 order:UIDocumentMenuOrderFirst
+                               handler:^{
+                                   [self chooseFromLibraryAsDocument];
+                               }];
```

Comment:
I didn't add separate "options" for Dropbox, iCloud as a) this appears non-trivial b) they are available via the "browse..." option.